server/api/utils/singletons/k8s.py

```diff
@@ -832,7 +816,22 @@ def get_secret_data(self, secret_name, namespace=""):
         secrets_data = self._get_secret_raw_data(secret_name, namespace)
         return self._decode_secret_data(secrets_data)
 
-    def _decode_secret_data(self, secrets_data, secret_keys=None):
+    def list_object_events(
+        self, object_name: str, namespace: str = None
+    ) -> list[client.CoreV1Event]:
+        return self._list_events(
+            namespace=namespace, field_selector=f"involvedObject.name={object_name}"
+        )
+
+    @raise_for_status_code
+    def _list_events(self, namespace=None, field_selector=""):
+        resp = self.v1api.list_namespaced_event(
+            self.resolve_namespace(namespace), field_selector=field_selector
+        )
+        return resp.items
```

Comment:
Why a wrapper for a private method?

Reply:
you mean decorator? It can be moved to `list_object_events` though I don't see a reason at this time

Reply:
I meant wrapper, as `list_object_events` doesn't do anything other than calling `_list_events`.
So are two functions needed here and not just have the implementation in `list_object_events`?

Reply:
It's not needed but it's to allow reusing `_list_events` without forcing the field selector.
There are other jira tickets that will use this in the future.